src/phpDocumentor/Reflection/Php/Factory/File.php

```diff
@@ -61,6 +65,30 @@ public function __construct(NodesFactory $nodesFactory, Adapter $adapter = null)
 
         $this->nodesFactory = $nodesFactory;
         $this->adapter = $adapter;
+
+        $this->middlewareChain = $this->createExecutionChain($middleware);
+    }
+
+    /**
+     * @param Middleware[] $middlewareList
+     *
+     * @return callable
+     */
+    private function createExecutionChain($middlewareList)
```

Comment:
This feels like a really generic thing that might be better as a separate class or factory function of an abstract Middleware class?